### PR TITLE
restore uvloop.new_event_loop and other missing uvloop members to typing

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: uvloop/__init__.pyi [hunk 1]
```diff
@@ -5,10 +5,25 @@ from collections.abc import Callable, Coroutine
...
 
-
 _T = TypeVar('_T')
 
 
+class Loop(AbstractEventLoop):
+    ...
+
+
+def new_event_loop() -> Loop:
```
Comment (graingert) [inline uvloop/__init__.pyi:15]: this is needed for `async with asyncio.Runner(loop_factory=uvloop.new_event_loop)`

File: uvloop/__init__.pyi
Comment (graingert) [inline uvloop/__init__.pyi:1]: adding this file removed the useful members like 'new_event_loop' which is needed for `asyncio.Runner(...)`

File: uvloop/__init__.py [hunk 2]
```diff
@@ -9,10 +11,12 @@
 from .loop import Loop as __BaseLoop  # NOQA
 from ._version import __version__  # NOQA
 
-
 __all__ = ('new_event_loop', 'install', 'EventLoopPolicy')
```
Comment (graingert) [inline uvloop/__init__.py:14]:
```suggestion

__all__ = ('new_event_loop', 'install', 'EventLoopPolicy')
```

File: uvloop/__init__.py [hunk 3]
```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
+
```
Comment (graingert) [inline uvloop/__init__.py:2]:
```suggestion
```

File: uvloop/__init__.py [hunk 4]
```diff
@@ -13,6 +13,9 @@
 __all__ = ('new_event_loop', 'install', 'EventLoopPolicy')
 
 
+_T = TypeVar("_T")
```
Comment (graingert) [inline uvloop/__init__.py:16]:
```suggestion
_T = _typing.TypeVar("_T")
```

File: uvloop/__init__.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import asyncio as __asyncio
 import typing as _typing
-import sys as _sys
```
Comment (graingert) [inline uvloop/__init__.py:3]: mypy doesn't recognise `_sys.version_info` in comparisons it needs the real `sys.version_info`

File: uvloop/__init__.py [hunk 6]
```diff
@@ -34,10 +37,16 @@ def install() -> None:
     __asyncio.set_event_loop_policy(EventLoopPolicy())
 
 
-def run(main, *, loop_factory=new_event_loop, debug=None, **run_kwargs):
+def run(
+    main: _typing.Coroutine[_typing.Any, _typing.Any, _T],
+    *,
+    loop_factory: _typing.Optional[_typing.Callable[[], Loop]] = new_event_loop,
+    debug: _typing.Optional[bool]=None,
+    **run_kwargs: object
```
Comment (graingert) [inline uvloop/__init__.py:45]: this is a bit problematic with mypy typing, it's a lie that run_kwargs exists but it's needed for backwards compatibility

Comment (graingert) [reply uvloop/__init__.py:45]: might be best to do this whole thing in an:

```
if _typing.TYPE_CHECKING:
    def run(...): ...
else:
    def run(..., **run_kwargs): ...
```

**Comment by graingert** [comment]
@fantix can I get a review on this please? It's blocking mypy in unicorn

File: uvloop/__init__.py [hunk 7]
```diff
@@ -34,69 +37,84 @@ def install() -> None:
...
-            raise ValueError("a coroutine was expected, got {!r}".format(main))
-
-        loop = loop_factory()
-        try:
-            __asyncio.set_event_loop(loop)
-            if debug is not None:
-                loop.set_debug(debug)
-            return loop.run_until_complete(wrapper())
-        finally:
+if _typing.TYPE_CHECKING:
```
Comment (fantix) [inline uvloop/__init__.py:40]: Can we just inline the typing and omit this TYPE_CHECKING check?

Comment (graingert) [reply uvloop/__init__.py:40]: I tried that, if you see the history of the PR, but it took too many changes

Comment (fantix) [reply uvloop/__init__.py:40]: I see now. What about adding `new_event_loop` to `__init__.pyi`?

**Review by fantix — APPROVED** [review]

File: uvloop/__init__.py (hunk 7 above)
Comment (graingert) [reply uvloop/__init__.py:40]: I tried that first also and it means the other members aren't type checked

**Pull request merged**